src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -189,25 +202,31 @@ private Task OnProjectFactoryCompletedAsync()
 
                 // Create new project context.
                 _currentAggregateProjectContext = await _contextProvider.Value.CreateProjectContextAsync().ConfigureAwait(false);
+                _currentTargetFramework = newTargetFramework;
 
                 // Dispose the old project context, if one exists.
                 if (previousContextToDispose != null)
                 {
-                    await _contextProvider.Value.ReleaseProjectContextAsync(previousContextToDispose).ConfigureAwait(false);
-
-                    foreach (var innerContext in previousContextToDispose.DisposedInnerProjectContexts)
-                    {
-                        foreach (var handler in Handlers)
-                        {
-                            await handler.Value.OnContextReleasedAsync(innerContext).ConfigureAwait(false);
-                        }
-                    }
+                    await DisposeAggregateProjectContextAsync(previousContextToDispose).ConfigureAwait(false);
```

Comment:
We shouldn't dispose the context until we know for sure we created a new context successfully. Regardless, if something throws during creating project context, LS will be busted regardless of what happens next.